applications/TrilinosApplication/custom_strategies/builder_and_solvers/trilinos_block_builder_and_solver.h

```diff
@@ -4,954 +4,830 @@
-    void CalculateReactions(
-        typename TSchemeType::Pointer pScheme,
-        ModelPart& r_model_part,
-        TSystemMatrixType& A,
-        TSystemVectorType& Dx,
-        TSystemVectorType& b) override
+    void CalculateReactions(typename TSchemeType::Pointer pScheme,
+                            ModelPart& rModelPart,
+                            TSystemMatrixType& rA,
+                            TSystemVectorType& rDx,
+                            TSystemVectorType& rb) override
     {
+        TSparseSpace::SetToZero(rb);
 
-        TSparseSpace::SetToZero(b);
-
-        //refresh RHS to have the correct reactions
-        BuildRHS(pScheme, r_model_part, b);
+        // refresh RHS to have the correct reactions
+        BuildRHS(pScheme, rModelPart, rb);
 
-        //initialize the Epetra importer
-        // TODO: this part of the code has been pasted until a better solution is found
-        int system_size = TSparseSpace::Size(b);
+        // initialize the Epetra importer
+        // TODO: this part of the code has been pasted until a better solution
+        // is found
+        int system_size = TSparseSpace::Size(rb);
         int number_of_dofs = BaseType::mDofSet.size();
-        std::vector< int > index_array(number_of_dofs);
+        std::vector<int> index_array(number_of_dofs);
 
-        //filling the array with the global ids
+        // filling the array with the global ids
         int counter = 0;
-        for(typename DofsArrayType::iterator i_dof = BaseType::mDofSet.begin(); i_dof != BaseType::mDofSet.end(); ++i_dof)
-        {
-            int id = i_dof->EquationId();
-            if( id < system_size )
-            {
-                index_array[counter] = id;
-                counter += 1;
+        int id = 0;
+        for (const auto& dof : BaseType::mDofSet) {
+            id = dof.EquationId();
+            if (id < system_size) {
+                index_array[counter++] = id;
             }
         }
 
-        std::sort(index_array.begin(),index_array.end());
-        std::vector<int>::iterator NewEnd = std::unique(index_array.begin(),index_array.end());
-        index_array.resize(NewEnd-index_array.begin());
+        std::sort(index_array.begin(), index_array.end());
+        std::vector<int>::iterator NewEnd =
+            std::unique(index_array.begin(), index_array.end());
+        index_array.resize(NewEnd - index_array.begin());
 
         int check_size = -1;
         int tot_update_dofs = index_array.size();
-        b.Comm().SumAll(&tot_update_dofs,&check_size,1);
-        if ( (check_size < system_size) &&  (b.Comm().MyPID() == 0) )
-        {
-            KRATOS_ERROR << "Dof count is not correct. There are less dofs than expected.\n"
-                         << "Expected number of active dofs = " << system_size << " dofs found = " << check_size ;
-        }
+        rb.Comm().SumAll(&tot_update_dofs, &check_size, 1);
+        KRATOS_ERROR_IF(check_size < system_size)
```

Comment:
this should be thrown on processor 1, otherwise we will have millinos of errors

Reply:
`KRATOS_ERROR_IF` (and all "standard" macros) only print from rank 0 in MPI. There are new `_ON_ALL_RANKS` versions to print from all ranks.

Reply:
So no change here !

Reply:
Actually, my last message was wrong (it is only valid for the logging, but errors do not use the logger). Still, I think it is healthy to throw from all ranks (otherwise other processes are kept alive waiting on the closest collective operation).